submissions/YuraZagor/Frogger/app.js

```diff
@@ -0,0 +1,109 @@
+const allEnemies = [];
+const speedLow = 200;
+const speedMedium = 275;
+const speedHigh = 350;
+const allSpeeds = [speedLow, speedMedium, speedHigh];
+const enemyRestartX = -55;
+const collisionOffset = 40;
+const cellX = 101;
+const cellY = 83;
+const spriteRepositioningY = -10;
+const playerStartX = 2*cellX;
+const playerStartY = 5*cellY + spriteRepositioningY;
+const fieldWidth = 5*cellX;
+const fieldHight = 5*cellY;
+
+const row1 = cellY + spriteRepositioningY;
+const row2 = 2*cellY + spriteRepositioningY;
+const row3 = 3*cellY + spriteRepositioningY;
+const rangesY = [row1, row2, row3];
+
+
+function Character(x, y, sprite) {
+    this.x = x;
+    this.y = y;
+    this.sprite=sprite;
+};
+
+Character.prototype.render = function() {
+    ctx.drawImage(Resources.get(this.sprite), this.x, this.y) 
+};
+
+function Enemy(x, y, sprite, speedX) {
+    Character.call(this, x, y, sprite);
+    this.speedX = speedX;
+};
+
+Enemy.prototype = Object.create(Character.prototype);
+
+Enemy.prototype.update = function(dt){
+    this.x += this.speedX * dt;
+    if (this.x > fieldWidth){
+        this.x = enemyRestartX;
+        this.speedX = allSpeeds[Math.floor(Math.random()*3)];
+    };
+    this.checkColision()
+};
+
+Enemy.prototype.checkColision = function(){
+    if (this.y === Player.y){
+        if ((this.x - Player.x < collisionOffset ) && (Player.x - this.x < collisionOffset )) {
+            Player.x = playerStartX;
+            Player.y = playerStartY;
+        };    
```

Comment:
Before we had here a reference to an instance of `Player` class stored in a global variable.
Now we directly refer here to the class itself.
Check the difference between class and an instance of a class.
Class is a blue-print. Enemies are intended to interact with specific instance (or instances in multiplayer mode) of `Player` class.
Also demo doesn't seem to work.

Reply:
not sure that I fully understand it all, but as I get it I should pass the instance as an attribute, so  that if we change the passed object we can still use the same method. Hope that was it

Reply:
> ...pass the instance as an **attribute**...
As an **argument**. Just for the sake of correct use of terminology